Demo not working for users outside uberVU

RadValentin opened the pull request:
Made it so when no labels are provided, the plugin will show all issues.

cc: @skidding @catalinmiron

Commit:
add functionality to show issues without labels #17

Commit:
fix spelling in variable name #17

Commit:
show issues without label in demo #17

Commit:
remove option to show issues without label #17

Commit:
remove default github labels #17

removed default labels because they were used when passing in 1 or 2
labels on plugin init

Commit:
show all issues when no labels are provided #17

Commit:
ignore codekit config files #17

Commit:
add gif showing the flow #17

RadValentin commented:
I've finished updating the demo, even added a nice gif to it :)
@skidding please review and let me know if there's anything else we can improve

ovidiuch commented on js/changelog.js:
```diff
@@ -6,7 +6,7 @@
     autoRefresh: false,
     githubRepo: 'uberVU/github-changelog-playground',
-    githubLabels: ['bug', 'enhancement', 'feature'],
+    githubLabels: [],
```
The default should use labels, we'll just set githubLabels to [] in the demo

RadValentin replied:
The reason I removed them was that if you initialized the plugin and set `githubLabels` to `[]` or `[bug]`(a single label) then the jQuery extend method would set it to `['bug', 'enhancement', 'feature']`. We can get around this by removing the deep extend [here](https://github.com/uberVU/github-changelog/blob/master/js/changelog.js#L32). This way whatever we pass in will override the default option.
What do you think?

ovidiuch commented on js/changelog.js:
```diff
@@ -122,16 +122,18 @@
+          $label,
           $title = $('<p>', {text: issue.title});
-
+      if(featuredLabel){
```
Look at @NiGhTTraX's pull request to see how to style this correctly.

ovidiuch commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+
+*.codekit
```
As I said to @NiGhTTraX as well, I like to keep project-related rules in the project .gitignore and user-related rules in the [global gitignore](https://help.github.com/articles/ignoring-files#create-a-global-gitignore)

ovidiuch commented:
Looking good.
1. Added some in-line comments
2. Remove the step with the labels from the demo
3. I'd add some more space between the two sections

Commit:
Revert "ignore codekit config files #17"

This reverts commit 4476d958b3795ca9d7784a35f6bf5a2253481b49.

Commit:
refactor addUpdateToList for better readability #17

Commit:
remove step 2 from demo #17

remove step stating that users should set labels in the playground as
github doesn’t allow this

Commit:
increase spacing between demo sections #17

RadValentin commented:
Almost done here, just need to clarify things regarding `githubLabels`(see inline comment) and we should be good.

Commit:
add empty labels array to demo #17

Commit:
remove deep extend on plugin options #17

removed it so that githubLabels can accept an empty array and so that
the default options won’t override this

Commit:
Revert "remove default github labels #17"

This reverts commit e42d80b7ac75a2263b26096680da799afa3692f4.

ovidiuch committed:
Merge branch 'master' into 17-demo-not-working-outside-ubervu

Conflicts:
	js/changelog.js

ovidiuch committed:
Only generate label when it exists #17
